File: DataBinding/PageAndRDLX/JsonDataSource/VB.NET/Web Service/CustomersController.vb
Imports Microsoft.AspNetCore.Mvc
Imports Microsoft.AspNetCore.Authorization
Imports Microsoft.AspNetCore.Hosting

<ApiController>
<Authorize(AuthenticationSchemes := "BasicAuthentication")>
<Route("[controller]/[action]")>
Public Class Customers
	Inherits ControllerBase
	Private ReadOnly _env As IWebHostEnvironment
	
	Sub New (env As IWebHostEnvironment)
		_env = env
	End Sub
	
	<HttpGet>
	Public Function GetJson() As Task (Of IActionResult)
		Return Task.FromResult (Of IActionResult)(new PhysicalFileResult(_env.ContentRootPath + $"/data/customers.json", "application/json"))
	End Function
End Class

File: DataBinding/PageAndRDLX/JsonDataSource/C#/Web Service/Controllers/CustomersController.cs
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace WebService.Controllers;

[ApiController]
[Authorize(AuthenticationSchemes = nameof(BasicAuthentication))]
[Route("[controller]/[action]")]
public class CustomersController : ControllerBase
{
	private readonly IWebHostEnvironment _env;
	public CustomersController(IWebHostEnvironment env)
	{
		_env = env;
	}
	
	[HttpGet]
	public Task<IActionResult> GetJson() =>
		Task.FromResult<IActionResult>(new PhysicalFileResult(_env.ContentRootPath + @"/data/customers.json", "application/json"));
}